veles/secrets/hcp/detector.go

```diff
@@ -36,108 +37,37 @@ var (
 	reClientSec = regexp.MustCompile(`["']?\b(?:HCP_CLIENT_SECRET|hcp_client_secret)\b["']?\s*[:=]\s*["']?([A-Za-z0-9._~\-]{64})["']?`)
 )
 
-// match holds the start offset and captured value of a regex match.
-type match struct {
-	start int
-	value string
-}
-
-// findMatches returns the start offsets and captured group values for all matches of re in data.
-func findMatches(re *regexp.Regexp, data []byte) []match {
-	idxs := re.FindAllSubmatchIndex(data, -1)
-	if len(idxs) == 0 {
-		return nil
-	}
-	out := make([]match, 0, len(idxs))
-	for _, m := range idxs {
-		// m[0], m[1] are the full-match bounds; m[2], m[3] are the first capture group bounds
-		out = append(out, match{start: m[2], value: string(data[m[2]:m[3]])})
+// NewPairDetector returns a Detector that finds HCP client credentials from key/value pairs.
+func NewPairDetector() veles.Detector {
+	return &pair.Detector{
+		MaxLen: maxPairWindowLen,
+		FindA:  findMatches(reClientID), FindB: findMatches(reClientSec),
+		FromPair: func(p pair.Pair) (veles.Secret, bool) {
+			return ClientCredentials{ClientID: p.A.Value, ClientSecret: p.B.Value}, true
+		},
+		FromPartialPair: func(p pair.Pair) (veles.Secret, bool) {
+			if p.A == nil {
```

Comment:
nit:  Just returning `ClientCredentials{ClientID: p.A.Value, ClientSecret: p.B.Value}, true` without a branch should result in the same behavior (not setting a value is the same as setting it to nil)